packages/x-scheduler/src/material/timeline/content/TimelineContent.tsx

```diff
@@ -0,0 +1,135 @@
+  months: 'months',
+  years: 'years',
+};
+
+export const TimelineContent = React.forwardRef(function TimelineContent(
+  props: TimelineContentProps,
+  forwardedRef: React.ForwardedRef<HTMLDivElement>,
+) {
+  const { className, ...other } = props;
+  const store = useTimelineStoreContext();
+  const resources = useStore(store, selectors.resources);
+  const visibleDate = useStore(store, selectors.visibleDate);
+  const view = useStore(store, selectors.view);
+
+  const start = visibleDate;
+  const end = React.useMemo(() => getEndBoundaries(view, start), [view, start]);
+
+  const occurrencesMap = useEventOccurrencesGroupedByResource({
+    start,
+    end,
+  });
+
+  const diff = diffIn(adapter, end, start, UNIT[view]);
+
+  const header = React.useMemo(() => {
+    switch (view) {
+      case 'days':
+        return <DaysHeader />;
+      case 'time':
+        return <TimeHeader />;
+      case 'weeks':
+        return <WeeksHeader />;
+      case 'months':
+        return <MonthsHeader />;
+      case 'years':
+        return <YearHeader />;
+      default:
+        return null;
+    }
+  }, [view]);
+
+  return (
+    <div className="TimelineViewContent" ref={forwardedRef} {...other}>
+      <TimelinePrimitive.Root
+        items={resources}
+        className="TimelineRoot"
+        style={
+          {
+            '--unit-count': diff,
+            '--unit-width': `var(--${view}-cell-width)`,
+            '--row-count': resources.length,
+          } as React.CSSProperties
+        }
+      >
+        <div className="TitleSubGridContainer">
+          <TimelinePrimitive.Row className="HeaderTitleRow">
+            <TimelinePrimitive.Cell className={clsx('TimelineCell', 'HeaderTitleCell')}>
+              Resource title
+            </TimelinePrimitive.Cell>
+          </TimelinePrimitive.Row>
+          <TimelinePrimitive.SubGrid className="TitleSubGrid">
+            {(resource) => (
+              <TimelinePrimitive.Row key={resource.name} className="TimelineRow">
+                <TimelinePrimitive.Cell className={clsx('TimelineCell', 'TimelineTitleCell')}>
+                  <span
+                    className={clsx(
+                      'ResourceLegendColor',
+                      getColorClassName(resource.eventColor ?? DEFAULT_EVENT_COLOR),
+                    )}
+                  />
+
+                  {resource.name}
+                </TimelinePrimitive.Cell>
+              </TimelinePrimitive.Row>
+            )}
+          </TimelinePrimitive.SubGrid>
+        </div>
+        <div className="EventSubGridContainer">
+          <TimelinePrimitive.Row className="HeaderRow">
+            <TimelinePrimitive.Cell className={clsx('TimelineCell', 'HeaderCell')}>
+              {header}
+            </TimelinePrimitive.Cell>
+          </TimelinePrimitive.Row>
+          <TimelinePrimitive.SubGrid className="EventSubGrid">
+            {Array.from(occurrencesMap.entries())?.map(([resourceId, occurrences]) => (
```

Comment:
We only use `occurrencesMap` as an array, so maybe we should change the signature of `useEventOccurrencesGroupedByResource` to return { resourceId: string, occurrences: CalendarEventOccurrence[]` instead.

It would also allow us to have a reliable sorting ot the resources (sort by resource name can be a good start).

Reply:
let me know id the current solution is what you meant here